cpp/src/strings/convert/convert_datetime.cu

```diff
@@ -208,32 +214,71 @@ struct parse_datetime {
 
       if (item.item_type == format_char_type::literal) {
         // static character we'll just skip;
-        // consume item.length bytes from string
+        // consume item.length bytes from the input string
         ptr += item.length;
         length -= item.length;
         continue;
       }
 
+      size_type bytes_read = item.length;  // number of bytes processed
       // special logic for each specifier
       switch (item.value) {
-        case 'Y': timeparts.year = static_cast<int16_t>(str2int(ptr, item.length)); break;
+        case 'Y': {
+          auto const [year, left] = parse_int(ptr, item.length);
+          timeparts.year          = static_cast<int16_t>(year);
+          bytes_read -= left;
+          break;
+        }
         case 'y': {
-          auto const year = str2int(ptr, item.length);
-          timeparts.year  = static_cast<int16_t>(year + (year < 69 ? 2000 : 1900));
+          auto const [year, left] = parse_int(ptr, item.length);
+          timeparts.year          = static_cast<int16_t>(year + (year < 69 ? 2000 : 1900));
```

Comment:
But what happens in 2069 😮

Reply:
[This](https://nvidia.slack.com/archives/CDTANRCTT/p1609862889009600?thread_ts=1609862482.009400&cid=CDTANRCTT) and then [this](https://nvidia.slack.com/archives/CDTANRCTT/p1609878276014900?thread_ts=1609862482.009400&cid=CDTANRCTT)